task/host-sp-comms/src/main.rs

```diff
@@ -924,6 +959,109 @@ impl ServerImpl {
         Ok(())
     }
 
+    fn handle_sprot(
+        &mut self,
+        sequence: u64,
+        command: HostToRotCommand,
+        data: &[u8],
+    ) {
+        match command {
+            HostToRotCommand::GetCertificates => {
+                let f = |buf: &mut [u8]| {
+                    let mut cnt = 0;
+                    let cert_chain_len = self
+                        .sprot
+                        .cert_chain_len()
+                        .map_err(AttestDataSprotError::from)?;
+                    let mut buf_idx = 0;
+                    for index in 0..cert_chain_len {
+                        let cert_len = self
+                            .sprot
+                            .cert_len(index)
+                            .map_err(AttestDataSprotError::from)?;
+                        for o in (0..cert_len).step_by(512) {
+                            let idx: usize = buf_idx + o as usize;
+                            let end = usize::min(512, (cert_len - o) as usize);
+
+                            self.sprot
+                                .cert(index, o, &mut buf[idx..idx + end])
```

Comment:
This will panic if `buf` is too small for all the certs, right? Should we guard against that and return an error instead? (Same comment in the closures below)

Reply:
That's unlikely to happen given the fixed size of the certs and the fixed size of the buffer for transfer based on the protocol. It might still be good to check because task crashes are not nice to debug.